packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -24,16 +24,60 @@ import 'code_signing.dart';
 import 'ios_workflow.dart';
 import 'mac.dart';
 
+/// [IOSDeploy] is responsible for spawning and managing `ios-deploy` processes.
+///
+/// When an app is launched via `ios-deploy`, we wait for the `ios-deploy` process
+/// until the app has been started via LLDB. Once it has started, `ios-deploy` puts LLDB
+/// into "autopilot" mode, and we continue.
+///
+/// If this (Flutter tools) process terminates before `ios-deploy` or the app, `ios-deploy`
+/// will be suspended. This is problematic because it will keep the app process on the phone
+/// alive via LLDB. Therefore we kill any previously created `ios-deploy` process which
+/// has launched an app before trying to run that app again.
 class IOSDeploy {
   const IOSDeploy();
 
+  File _tempFile(ApplicationPackage package) => fs.file('/tmp/${package.id}.ios_deploy.pid');
```

Comment:
I think you should be able to write a test case that verifies that the pid is written (using the memory filesystem) and can be read correctly.

Reply:
I've added a few tests to `devices_test.dart`.